vmcheck: allow ignoring iopl errors

lucab opened the pull request:
This introduces an additional flag for the VM-checking logic,
in order to allow consuming applications to keep going when changing
I/O access level is not alloewd (see `kernel_lockdown(7)` on Linux).
In most cases the library may work without higher I/O level,
and actual failures can be handled via normal error-handling on RPC
methods.

Ref: https://github.com/vmware/vmw-guestinfo/issues/20

lucab commented:
This is an hotfix which does not completely fix the reported bug, but it alleviates the symptoms in most cases. See https://github.com/vmware/vmw-guestinfo/issues/20#issuecomment-699899854 for all the details.

jlebon commented on vmcheck/vmcheck.go:
```diff
@@ -24,21 +24,26 @@ import (
+// It also tries to elevate I/O privileges for the calling thread, which in
+// some cases may be forbidden by the system (e.g Linux in `kernel_lockdown` mode
+// does not allow `iopl` calls); the `ignoreAccessErrors` parameter allows
+// to control library behavior in order to threat such errors as non-fatal.
```
```suggestion
// to control library behavior in order to treat such errors as non-fatal.
```

lucab replied:
Good catch! I've directly amended the typo in the existing commit.

jlebon reviewed (approved):
Seems reasonable to me!

lucab commented:
@frapposelli no problem, rebased.

Commit:
vmcheck: allow ignoring iopl errors

This introduces an additional flag for the VM-checking logic,
in order to allow consuming applications to keep going when changing
I/O access level is not alloewd (see `kernel_lockdown(7)` on Linux).
In most cases the library may work without higher I/O level,
and actual failures can be handled via normal error-handling on RPC
methods.

Signed-off-by: Luca BRUNO <luca.bruno@coreos.com>

lucab commented:
@frapposelli ping, any chance to have this reviewed and merged?

frapposelli commented:
@lucab sorry, this fell through the cracks. Merged.
